pennylane/_queuing_context.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the :class:`QueuingContext` abstract base class.
+"""
+import abc
+
+
+class QueuingContext(abc.ABC):
+    """Abstract base class for classes that exposes a queue for Operations.
```

Comment:
An extra sentence with an example could be helpful here. This one that was previously used for the `Recorder` is pretty good in my opinion:
`"The Recorder class is a very minimal QNode, that simply provides a QNode context for operator queueing."`

Overall, from my side something along the lines of answering how defining a `QueuingContext` comes in to play (e.g. for `QNodes`, operation recorders, etc.) would be perfect!

Reply:
Please see my take on it! Feel free to edit it tough, I'm not 100% happy.